components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -0,0 +1,226 @@
+
+	var configPath = ""
+	var err error
+	if len(args) > 0 {
+		configPath = args[0]
+	}
+
+	if verifyCmdFlags.haAWSProvision {
+		err = verifyHaAWSProvision(configPath)
+	}
+
+	if verifyCmdFlags.haAWSManagedProvision {
+		err = verifyHaAWSManagedProvision(configPath)
+	}
+
+	if verifyCmdFlags.haAWSDeploy {
+		err = verifyHaAWSDeploy(configPath)
+	}
+
+	if verifyCmdFlags.haAWSManagedDeploy {
+		err = verifyHaAWSManagedDeploy(configPath)
+	}
+
+	if verifyCmdFlags.standaloneDeploy {
+		err = verifyStandaloneDeploy(configPath)
+	}
+
+	if verifyCmdFlags.haOnpremDeploy {
+		err = verifyHaOnpremDeploy(configPath)
+	}
+
+	if verifyCmdFlags.haOnPremAWSManagedDeploy {
+		err = verifyHaOnPremAWSManagedDeploy(configPath)
+	}
+
+	if verifyCmdFlags.haOnPremCustManagedDeploy {
+		err = verifyHaOnPremCustManagedDeploy(configPath)
+	}
+
+	if verifyCmdFlags.certificates {
+		err = verification.VerifyCertificates("")
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func verifyHaAWSProvision(configPath string) error {
+	if isA2HARBFileExist() {
+		return status.New(status.InvalidCommandArgsError, "Setup is already Provisioned. Please use --ha-aws-deploy flag.")
+	} else {
+		err := verification.VerifyHAAWSProvision(configPath)
+		if err != nil {
+			return status.Annotate(err, status.ConfigError)
+		}
+	}
+	return nil
+}
+
+func verifyHaAWSManagedProvision(configPath string) error {
+	if isA2HARBFileExist() {
+		return status.New(status.InvalidCommandArgsError, "Setup is already Provisioned. Please use --ha-aws-managed-deploy flag.")
+	} else {
+		err := verification.VerifyHAAWSProvision(configPath)
+		if err != nil {
+			return status.Annotate(err, status.ConfigError)
+		}
+	}
+	return nil
+}
+
+func verifyHaAWSDeploy(configPath string) error {
+	if !isManagedServicesOn() && isA2HARBFileExist() {
+		err := verification.VerifyHAAWSDeployment(configPath)
+		if err != nil {
+			return status.Annotate(err, status.ConfigError)
+		}
+	} else if !isA2HARBFileExist() {
+		return status.New(status.InvalidCommandArgsError, errProvisonInfra)
+	} else {
+		return status.New(status.InvalidCommandArgsError, "This flag will not verify the Managed Services Setup. Please use the --ha-aws-managed-deploy flag.")
+	}
+	return nil
+}
+
+func verifyHaAWSManagedDeploy(configPath string) error {
+	if isManagedServicesOn() && isA2HARBFileExist() {
+		err := verification.VerifyHAAWSManagedDeployment(configPath)
+		if err != nil {
+			return status.Annotate(err, status.ConfigError)
+		}
+	} else if !isA2HARBFileExist() {
+		return status.New(status.InvalidCommandArgsError, errProvisonInfra)
+	} else {
+		return status.New(status.InvalidCommandArgsError, "Managed Services flag is not set. Cannot verify the config.")
+	}
+	return nil
+}
+
+func verifyStandaloneDeploy(configPath string) error {
+	if isA2HARBFileExist() {
+		return status.New(status.InvalidCommandArgsError, "Deployment type does not match with the requested flag.")
+	} else if !isA2HARBFileExist() {
```

Comment:
only `else` statement will be sufficient here, no need of `else if`

Reply:
Done